component/libfuse/libfuse.go

```diff
@@ -204,6 +207,23 @@ func (lf *Libfuse) Validate(opt *LibfuseOptions) error {
 		lf.negativeTimeout = defaultNegativeEntryExpiration
 	}
 
+	if lf.allowOther {
+		// Validate allow_other is enabled in fuse.conf file or not
+		cmd := exec.Command("bash", "-c", "cat /etc/fuse.conf | grep -i user_allow_other$")
```

Comment:
default value is disabled now, so using `allow_other` option would always return error now?
```
# cat /etc/fuse.conf
# /etc/fuse.conf - Configuration file for Filesystem in Userspace (FUSE)

# Set the maximum number of FUSE mounts allowed to non-root users.
# The default is 1000.
#mount_max = 1000

# Allow non-root users to specify the allow_other or allow_root mount options.
#user_allow_other
```

Reply:
cc @cvvz

Reply:
even today its failing, it just fails silently. With this change it will start returning non-zero error code.

Reply:
the above is the default setting in AKS node, and user could set `-o allow_other` mount option, and it won't return error on AKS node today

Reply:
`-o allow_other` only failed when mount by non-root user. Root user can mount successfully even without configuring `/etc/fuse.conf`.

Reply:
If user_allow_other is not enabled in /etc/fuse.conf then it shall not work irrespective of you execute mount with sudo or non-sudo or root user.
In current implementation it fails without giving any error and with this change we are pre-validating the file and returning error if mount is suppose to fail later due to this check.
One question here for the AKS team, if we always mount with allow_other option in AKS why dont we enable user_allow_other in /etc/fuse.conf by default when pod is deployed ?

Reply:
Ok I verified this and appears root user can mount with allow_other even then /etc/fuse.conf does not have this option set. So I can add a check for uid != 0 while checking the things.

Reply:
Corrected code to skip this check for root user